sys/arduino/arduino_sketches.inc.mk

```diff
@@ -0,0 +1,31 @@
+# Compile together the Arduino sketches of the application
+# They are declared a as new module $(SKETCH_MODULE) in $(SKETCH_MOD_DIR)
+
+ifndef SKETCH_MODULE
+  $(error SKETCH_MODULE undefined. It should be defined to the sketches module name)
+endif
+ifndef SKETCH_MODULE_DIR
+  $(error SKETCH_MODULE_DIR undefined. It should be defined to the sketches module directory)
+endif
+ifndef SKETCHES
+  $(error SKETCHES undefined. It should be defined to the list of sketches files)
+endif
+
+SNIPDIR         = $(RIOTBASE)/sys/arduino
+SKETCHES_ALL    = $(SNIPDIR)/pre.snip $(SKETCHES) $(SNIPDIR)/post.snip
+SKETCH_CPP     ?= arduino_sketches.cpp
+
+# Building the module files
+#   Do not use $^ in receipes as Makefile is also a prerequisite
+$(SKETCH_MODULE_DIR)/Makefile: $(SKETCH_MODULE_DIR)/$(SKETCH_CPP)
+	$(Q)echo 'SRCXX = $(SKETCH_CPP)'               > $@
+	$(Q)echo 'include $$(RIOTBASE)/Makefile.base' >> $@
+$(SKETCH_MODULE_DIR)/$(SKETCH_CPP): $(SKETCHES_ALL) | $(SKETCH_MODULE_DIR)
+	$(Q)cat $(SKETCHES_ALL) > $@
+$(SKETCH_MODULE_DIR):
+	@mkdir -p $@
+
+# Make everything rebuild if current makefile changes
+_ARDUINO_SKETCHES_MAKEFILE = $(lastword $(MAKEFILE_LIST))
```

Comment:
I think this needs an immediate assignment (`:=`)to avoid the situation where MAKEFILE_LIST is updated by including another file later.

Reply:
The value is used directly and only in the two following lines, so following imports will not impact.
However, as I am not using immediate assignment, `lastword` is executed twice, so I will change.